tests/hgetall_test.go

```diff
@@ -0,0 +1,58 @@
+package tests
+
+import (
+	"gotest.tools/v3/assert"
+	"reflect"
+	"testing"
+)
+
+func TestHGETALL(t *testing.T) {
+	conn := getLocalConnection()
+	defer conn.Close()
+
+	testCases := []TestCase{
+		{
+			commands: []string{"HSET key_hGetAll field value", "HSET key_hGetAll field2 value_new", "HGETALL key_hGetAll"},
+			expected: []interface{}{ONE, ONE, []string{"field", "value", "field2", "value_new"}},
+		},
+		{
+			commands: []string{"HGETALL key_hGetAll01"},
+			expected: []interface{}{[]interface{}{}},
+		},
+		{
+			commands: []string{"SET key_hGetAll02 field", "HGETALL key_hGetAll02"},
+			expected: []interface{}{"OK", "WRONGTYPE Operation against a key holding the wrong kind of value"},
+		},
+		{
+			commands: []string{"HGETALL key_hGetAll03 x", "HGETALL"},
+			expected: []interface{}{"ERR wrong number of arguments for 'hgetall' command",
+				"ERR wrong number of arguments for 'hgetall' command"},
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
Run cleanup logic using defer to cleanup the keys after the test is done. That may also be the reason why your tests were failing earlier. All integration tests share the same server, therefore each test needs to ensure it cleans up its resources once its done.

Reply:
yes that was exactly the reason, i already mentioned it over here https://github.com/DiceDB/dice/pull/431#issuecomment-2323299209.
As of now I have deleted keys in defer..
but we may need to find better approach.